File: src/main/scala/RegistryMessages.scala
package codecraft.registry



final case class CmdRoute(
  urlPath: String,
  cmdKey: String
)

final case class Registry(
  id: String,
  active: Boolean,
  protocol: String,
  routes: List[CmdRoute],
  host: String,
  port: Int
)

final case class PostRegistry(
  active: Boolean,
  protocol: String,
  routes: List[CmdRoute],
  host: String,
  port: Int
)

final case class DeleteRegistry(
  id: String
)

final case class DeleteRegistryReply(
  code: Int,
  status: Option[String]
)

final case class PostRegistryReply(
  id: Option[String],
  code: Int,
  status: Option[String]
)

final case class GetRegistries(
  protocol: String,
  cmdKey: String
)

final case class GetRegistriesReply(
  registries: Option[List[Registry]],
  code: Int,
  status: Option[String]
)

final case class RegistryUpdated(
  id: String,
  registry: Registry,
  active: Boolean
)

File: src/main/fsharp/RegistryMessages.fs
namespace codecraft.registry



type CmdRoute = {
  urlPath: string
  cmdKey: string
}

type Registry = {
  id: string
  active: bool
  protocol: string
  routes: list<CmdRoute>
  host: string
  port: int
}

type PostRegistry = {
  active: bool
  protocol: string
  routes: list<CmdRoute>
  host: string
  port: int
}

type DeleteRegistry = {
  id: string
}

type DeleteRegistryReply = {
  code: int
  status: option<string>
}

type PostRegistryReply = {
  id: option<string>
  code: int
  status: option<string>
}

type GetRegistries = {
  protocol: string
  cmdKey: string
}

type GetRegistriesReply = {
  registries: option<list<Registry>>
  code: int
  status: option<string>
}

type RegistryUpdated = {
  id: string
  registry: Registry
  active: bool
}